server/state/index.js

```diff
@@ -23,21 +23,30 @@ const initSmartContracts = () => {
   smartContracts = List()
 }
 
+const isDuplicateSmartContract = smartContract => {
+  if (smartContracts.find(x => x.smartContract === smartContract)) {
+    return true
+  } else return false
+}
+
 const addSmartContract = async smartContractObj => {
   const invalidFields = findInvalidSmartContractFields(smartContractObj)
-  if (invalidFields.length > 0)
+  if (invalidFields.length > 0) {
     throw new Error(
```

Comment:
Same thing here with the comment below - if we want to create errors here we should be creating an error handling middleware and passing the error along the route handlers using `next(error)`

Here I think we send a `400` bad request back to the client. Maybe we do this in a separate PR?